src/MICore/Transports/UnixShellPortTransport.cs

```diff
@@ -34,12 +39,69 @@ public void Init(ITransportCallback transportCallback, LaunchOptions options, Lo
             _logger = logger;
             _startRemoteDebuggerCommand = _launchOptions.StartRemoteDebuggerCommand;
 
-            waitLoop?.SetText(MICoreResources.Info_InstallingDebuggerOnRemote);
+            if (_launchOptions.DebuggerMIMode == MIMode.Clrdbg)
+            {
+                if (!UnixShellPortLaunchOptions.HasSuccessfulPreviousLaunch(_launchOptions))
+                {
+                    waitLoop?.SetText(MICoreResources.Info_InstallingDebuggerOnRemote);
+                    try
+                    {
+                        Task.Run(async () => await DownloadAndCopyFileToRemote(_launchOptions.DebuggerInstallationDirectory, _launchOptions.GetClrDbgUrl)).Wait();
```

Comment:
>async () => await [](start = 33, length = 17)

nit: you don't need the 'async' / 'await' part of this as DownloadAndCopyFileToRemote already returns a Task